ouroboros-consensus/ouroboros-consensus-test-infra/src/Test/ThreadNet/Network.hs

```diff
@@ -116,25 +117,35 @@ import           Test.Util.Tracer
 
 -- | How to forge an EBB
 --
-type ForgeEBB blk =
-     TopLevelConfig blk
-  -> SlotNo          -- ^ EBB slot
-  -> BlockNo         -- ^ EBB block number (i.e. that of its predecessor)
-  -> ChainHash blk   -- ^ EBB predecessor's hash
-  -> blk
+data ForgeEbbEnv blk = ForgeEbbEnv
+  { forgeEBB ::
+         TopLevelConfig blk
+      -> SlotNo
+         -- EBB slot
+      -> BlockNo
+         -- EBB block number (i.e. that of its predecessor)
+      -> ChainHash blk
+         -- EBB predecessor's hash
+      -> blk
+  , ebbSlotBefore :: SlotNo -> SlotNo
+    -- ^ the slot of the most recent expected that precedes the given slot
```

Comment:
I don't understand this name or this comment :)

Reply:
Elaborated in commit 9b2aa86f61b252315479113627e679953d02c542